src/parser.rs

```diff
@@ -84,18 +84,31 @@ pub fn parse_output(tokens: &[Token]) -> Result<Output> {
         }
 
         try!(expect(&mut iter, Colon));
+        let mut comma_previous = false;
+        let mut first = true;
 
         // loops through the argument list after the filter name
         while iter.peek() != None && iter.peek().unwrap() != &&Pipe {
             match iter.next().unwrap() {
-                &Comma => continue, // next argument
+                &Comma => {
+                    comma_previous = true;
+                    continue // next argument
+                }
                 x @ &StringLiteral(_) |
                 x @ &NumberLiteral(_) |
-                x @ &BooleanLiteral(_) => args.push(try!(Value::from_token(x))),
+                x @ &BooleanLiteral(_) => args.push(VarOrVal::Val(try!(Value::from_token(x)))),
+                &Identifier(ref v) => {
+                    if !comma_previous && !first {
+                        return Error::parser("a comma or a pipe", Some(&Token::Identifier(v.clone())));
+                    }
+                    args.push(VarOrVal::Var(Variable::new(v)))
+                }
                 x => {
                     return Error::parser("a comma or a pipe", Some(x));
                 }
             }
+            comma_previous = false;
+            first = false;
```

Comment:
I don't really like this piece.

Essentially, a test was failing because instead of raising an error for a missing comma or pipe, the parser was eating the next identifier, producing the wrong result.

To get around this, I'm keeping track of whether or not this is the first time through the list, and whether the previous token was a comma. If a comma was not directly before the new Identifier, and its not the first time through the loop, then we raise an error. This fixes the test.

A better way to do this would probably be to look ahead after parsing the Identifier and check if the next token is either a comma or a pipe. What do you think? Any thoughts / suggestions?

Let me know if you need me to clarify. If you want to see the failing test, just revert the second commit in this PR.